### ISSUE TITLE
Combining efforts

### ISSUE THREAD

**Issue by balapal** [mrchimp/imagelightbox#3]
I have found a few forks/initiatives/repositories based on http://osvaldas.info/image-lightbox-responsive-touch-friendly
Namely:
https://github.com/felipearosemena/imageLightbox
https://github.com/rejas/imagelightbox
https://github.com/mrchimp/imagelightbox
https://github.com/jeremydouglas/imagelightbox

I was wondering if you would be open combining those efforts?
I am raising this topic as a new issue with all of the above repos (not sure if there is any better way getting in touch with the authors).

**Comment by mrchimp** [comment]
Sure. Sounds good to me.

**Comment by balapal** [comment]
Cool. I will check how much each of them differ.
We will have to manually merge the different version I guess.
Any ideas how we could manage it effectively?

**Comment by rejas** [comment]
Combining efforts of course would be nice. Not sure though how to determine where to do it best, dont know about how github manages "collaborations", until now I was more of a lonely coder :-D

**Comment by mrchimp** [comment]
Well I guess the best way is to pick one of the repos as the main one (or create a new one) and then merge features from the others into it bit by bit. It's going to take some work as they seem to have diverged from the source quite a bit... We can add collaborators to the repo so it's not just down to one person to keep on top of the pull requests. Github's great for collaborating but there's not really anywhere to talk except here in the issue tracker. We could use [Slack](http://slack.com) maybe? We use it at work and it works well.

I have to say I'm not as excited about imagelightbox as I was when I first found it... It's still good and it serves its purpose well but I don't know how long its lifespan is. I mean, it might be better to rewrite it as a web component but I don't know if that's what people want... I dunno, I'm rambling here but what are your thoughts? Do we just merge them together and maintain it as a jquery plugin? Do we try to decouple jquery? Do we remake it as a web component? Bower or NPM? Tea or coffee?

**Comment by balapal** [comment]
My initial goal is to merge into a single source code the bug fixes and improvements already out there in the various forks.
So there would be a golden source containing a better version than the current individual repositories and where people can contribute to.

If somebody wants to rewrite it later in a more modern way that would be also awesome.

Let's decide first which should be the main repository.

Here is the documentation of how to add collaborators:
https://help.github.com/articles/adding-collaborators-to-a-personal-repository/
https://help.github.com/articles/what-are-the-different-access-permissions/

**Comment by rejas** [comment]
some thoughts from me:
1) love the idea of a web component, wanted to look into that for a long time..
2) i am already using bower in my repo, since we are doing frontend stuff here :-)
3) why merging into an existing repo? the beauty of git is that you can make your own fork and pull the stuff from other repos into it. if we collaborate we would probably have still our own repos for preparing stuff and then push it into your "main"-repo. or do i miss something here?

**Comment by balapal** [comment]
Re. 3) You are right. What I meant that there are these independent repositories that may contain some bug fixes that we want to merge into our main repo.

**Comment by balapal** [comment]
I have compared https://github.com/rejas/imagelightbox/blob/master/src/imagelightbox.js with https://github.com/felipearosemena/imageLightbox/blob/master/imageLightbox.js

The only material difference I found is that in line that rejas's version returns false and not true in setImage() function. 

``` javascript
setImage = function() {
            if (!image.length) return true;
```

**Comment by mrchimp** [comment]
I agree that Bower is best for frontend but some people like using NPM for everything. There's no reason we can't make it available on both.

I'm not sure that it makes sense to maintain a jquery plugin version and a web component in a single package.

As for the existing repo thing... Whatever we do, we should designate a repo as the main one, whether that's an existing one or a newly forked one. Then we can merge into that from wherever. We could maintain our own separate ones and pull others' changes in as we see fit but if our goal is collaboration  and getting the most from all of the different sources then it makes sense to all work on a single package.

I'm happy to add you guys to my repo as collaborators and use that as the main one. Then we could fork that into a separate web component repo?

**Comment by balapal** [comment]
I agree, It would make sense to use one of the existing repos as the main one.
@rejas's repo seems to be more active in terms of contributors, forks, commits, stars. Therefore I think it would be the best as the main repo.
What do you think guys?

**Comment by mrchimp** [comment]
Sounds good. I think we should draw up a list of features that need merging then set up issues on the repo so we can work through them and keep track of our progress. Then when the features have been merged, create a new fork for the web component version.

**Comment by balapal** [comment]
I have went through the commits of https://github.com/felipearosemena/imageLightbox
It's an example of how to use the original version of imageLightbox.
The only difference is the one I mentioned 4 comments before. Not sure if we want to merge that.

**Comment by balapal** [comment]
I have went through the commits of https://github.com/felipearosemena/imageLightbox
It's an example of how to use the original version of imageLightbox.
The only difference is the one I mentioned 4 comments before. Not sure if we want to merge that.

**Comment by rejas** [comment]
Not sure about why that return value was false in my repo, maybe the original code was different and got cahnged to true someday?

**Comment by mrchimp** [comment]
This can wait until later on but we should set up a demo using Github pages.

**Comment by balapal** [comment]
I am thinking of checking with the original author if he is interested in joining maintaining the library on github. http://osvaldas.info/about

**Comment by balapal** [comment]
@rejas http://osvaldas.info/examples/image-lightbox-responsive-touch-friendly/imagelightbox.js returns true.

**Comment by mrchimp** [comment]
I've sent him a message on Twitter.

**Comment by rejas** [comment]
@balapal changed it (although I dont think it makes much difference :-)

**Comment by rejas** [comment]
you guys any idea on how to handle the bower registry? mrchimp already has imageligthbox registered and I dont know what other name i could use for the library :-(

**Comment by balapal** [comment]
Haven't used bower before. I am mainly a Java guy :)

**Comment by mrchimp** [comment]
Hmm. I wouldn't mind surrendering the name except that I have projects
depending on it which I'd rather not break unexpectedly...

How about imagelightbox2?
jquerylightbox?
nicelightbox?
betterlightbox?

**Comment by rejas** [comment]
well, imagelightbox2 sounds good to me :-)
